backend/src/post/post.service.ts

```diff
@@ -17,13 +25,25 @@ export class PostService {
     private userService: UserService,
   ) {}
 
-  async findAll(): Promise<Post[]> {
-    return this.postRepository.find({
-      order: {
-        createdAt: 'DESC',
-      },
-    });
+  async findAll(): Promise<IPost[]> {
+    return this.postRepository
+      .createQueryBuilder('post')
+      .leftJoinAndSelect('post.user', 'user') // Join the user entity
+      .addSelect('user.name') // Only select the user's name
+      .orderBy('post.createdAt', 'DESC') // Order posts by createdAt
+      .getMany()
+      .then((posts) =>
+        posts.map((post) => ({
+          id: post.id,
+          content: post.content,
+          createdAt: post.createdAt,
+          updatedAt: post.updatedAt,
+          userId: post.user.id,
+          user: post.user.name,
+        })),
+      );
```

Comment:
It is a very bad idea to add some maps/filters on the BE if they are unnecessary - here we can form the data in the request to the Postgres.

[Example](https://orkhan.gitbook.io/typeorm/docs/select-query-builder#getting-values-using-querybuilder):
```
const photosSums = await dataSource
    .getRepository(User)
    .createQueryBuilder("user")
    .select("user.id")
    .addSelect("SUM(user.photosCount)", "sum")
    .groupBy("user.id")
    .getRawMany()
```